src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2235 @@
+        gaussian_filter_9x9_row_hip_compute(&srcShared[hipThreadIdx_y_channel.z + 8][hipThreadIdx_x8], &sum_f24.f8[2], filter_row9);
+        rpp_hip_adjust_range(dstPtr, &sum_f24);
+        rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &sum_f24);
+    }
+}
+
+__device__ float gaussian(int iSquare, int j, float mulFactor)
+{
+    float expFactor = - (iSquare + (j * j)) * mulFactor;
+    expFactor = expf(expFactor);
+    return expFactor;
+}
+
+template <typename T, typename U>
+__global__ void create_gaussian_kernel(T *filterTensor,
+                                       U *rowType,
+                                       float *stdDevTensor,
+                                       int kernelSize,
+                                       int filterStride,
+                                       int batchSize)
+{
+    int id_x = (hipBlockIdx_x * hipBlockDim_x + hipThreadIdx_x);
+    if(id_x >= batchSize)
+        return;
+
+    T filter_temp;
+    T *filter = &filterTensor[id_x];
+    float stdDev = stdDevTensor[id_x];
+    float mulFactor = 1 / (2 * stdDev * stdDev);
+    int kernelSizeByTwo = kernelSize / 2;
+    int startIdx = - (kernelSizeByTwo);
+    int lastRowIdx = (kernelSize - 1) * kernelSize;
+    int rowIdx = 0;
+
+    // compute values for only top left quarter and replicate the values
+    for(int i = startIdx; i <= 0; i++, rowIdx += kernelSize)
+    {
+        int iSquare = i * i;
+        int colIdx = 0;
+        for(int j = startIdx; j <= 0; j++, colIdx++)
+        {
+            filter_temp.f1[rowIdx + colIdx] = gaussian(iSquare, j, mulFactor);
+            filter_temp.f1[rowIdx + kernelSize - 1 - colIdx] = filter_temp.f1[rowIdx + colIdx];
+        }
+
+        // Copy symmetric rows
+        if((lastRowIdx - rowIdx) != rowIdx)
+            *(U *)(&filter_temp.f1[lastRowIdx - rowIdx]) = *(U *)(&filter_temp.f1[rowIdx]); // To check, if branching inside if for loop cause degration in performance
+    }
+
+
+    // Find the sum of kernel values
+    int cnt = 0;
+    float kernelSum = 0.0f;
+    for(int i = startIdx; i <= kernelSizeByTwo; i++)
+    {
+        for(int j = startIdx; j <= kernelSizeByTwo; j++)
+        {
+            kernelSum += filter_temp.f1[cnt];
+            cnt++;
+        }
+    }
+    kernelSum = (1.0f / kernelSum);
+
+    // Multiply kernel values by (1 / kernelSum)
+    cnt = 0;
+    for(int i = startIdx; i <= kernelSizeByTwo; i++)
+    {
+        for(int j = startIdx; j <= kernelSizeByTwo; j++)
+        {
+            filter_temp.f1[cnt] *= kernelSum;
+            cnt++;
+        }
+    }
+
+    *(reinterpret_cast<T *>(filter)) = filter_temp;
+}
+
+static RppStatus hip_exec_create_gaussian_kernel(Rpp32f *filterTensor,
+                                                 Rpp32s kernelSize,
+                                                 Rpp32f *stdDevTensor,
+                                                 rpp::Handle &handle)
+{
+    int localThreads_x = 256;
```

Comment:
please get rid of constant assignments here and use it inside dims3 macro. For e.g, dim3{256}

Reply:
Addressed in #204